matrix-sdk/src/main/java/org/matrix/androidsdk/db/MXMediasCache.java

```diff
@@ -517,6 +534,47 @@ public void clearTmpCache() {
         }
     }
 
+    /**
+     * Move a decrypted media file to the /share folder, to avoid this file to be deleted if in the /tmp folder
+     *
+     * @param fileToMove The file to move
+     * @return The copied file in the Share folder location
+     */
+    public File moveToShareFolder(final File fileToMove,
+                                  final String filename) {
+        File dstFile = new File(mShareFolderFile, filename);
+
+        if (dstFile.exists()) {
+            if (!dstFile.delete()) {
+                Log.w(LOG_TAG, "Unable to delete file");
+            }
+        }
+
+        if (!fileToMove.renameTo(dstFile)) {
+            Log.w(LOG_TAG, "Unable to rename file");
+
+            // Return the original file
+            return fileToMove;
+        }
+
+        return dstFile;
+    }
+
+    /**
+     * Clear the temporary shared decrypted media cache folder
+     */
+    public void clearShareDecryptedMediaCache() {
+        Log.d(LOG_TAG, "clearShareDecryptedMediaCache()");
+
+        if (mShareFolderFile.exists()) {
+            ContentUtils.deleteDirectory(mShareFolderFile);
+        }
+
+        if (!mShareFolderFile.exists()) {
```

Comment:
Why do you test if the folder exist, you just deleted it ?

Reply:
I've copied what is done in `clearTmpDecryptedMediaCache()`...
I suppose `ContentUtils.deleteDirectory()` may fail for some reason.